tests/socket/socket_tests_common.h

```diff
@@ -18,6 +18,14 @@ typedef struct _ip_address
     };
 } ip_address_t;
 
+typedef enum _protocol_type
+{
+    INVALID,
+    TCP,
+    UDP,
+    CONNECTED_UDP
```

Comment:
"connected UDP" is not a separate "protocol" from "udp".
I'd suggest not using the word protocol for this, or making connected/unconnected a separate field from IPPROTO.  Furthermore, if you call something CONNECTED_UDP, then you should change UDP to UNCONNECTED_UDP.

Reply:
This comment is not resolved yet.  ("UDP" and "CONNECTED_UDP" are not alternatives.  The former term includes the latter.)